mathics/core/expression.py

```diff
@@ -1150,25 +1180,30 @@ def eval_range(indices):
                 eval_range(range(len(elements)))
                 # rest_range(range(0, 0))
 
+        # Step 2: Build a new expression. We take care not
+        # to evaluate elements, run to_python() on them in
+        # Expression construction, or convert Expresions elements from a tuple to a list
+        # and back if that can be avoided.
+
         if self._elements_fully_evaluated:
-            elements = self._elements
+            new = Expression(
+                head,
+                *self._elements,
+                element_properties={
+                    "_is_flat": self._is_flat,
+                    "_is_sorted": self._is_sorted,
+                    "_elements_fully_evaluated": self._elements_fully_evaluated,
+                }
+            )
+            elements = self.elements
         else:
             elements = self.get_mutable_elements()
             eval_elements()
-
-        # Step 2: Build a new expression. Notice that elements are given
-        # after creating the object, to avoid to call `from_python` on each element.
-
-        # FIXME: put this in a method
-        new = Expression(head)
-        new.elements = elements
-        # the setter in new.elements calls _build_elements, and it sets new._is_flat
-        # new._is_flat = self._is_flat
+            new = Expression(head, *elements)
 
         # Step 3: Now, process the attributes of head
         # If there are sequence, flatten them if the attributes allow it.
-        # if not new._is_flat and not (SEQUENCE_HOLD | HOLD_ALL_COMPLETE) & attributes:
```

Comment:
Last week this was failing. Probably as a result of the refactoring separating Box elements from general elements fixed this.

Reply:
we should avoid calling `evaluate` over Box elements. `Box` elements should be handled just as a `String`.